lib/composer/package/loader/project_loader.rb

```diff
@@ -0,0 +1,52 @@
+module Composer
+  module Package
+    module Loader
+      # Loads a package from existing configuration
+      class ProjectLoader
+
+        def initialize(loader = nil)
+          if loader.nil?
+            loader = Composer::Package::Loader::HashLoader.new
+          end
+          @loader = loader
+        end
+
+        # Load a project ref as a package
+        # Param:  Project project The target gitlab project.
+        # Param:  Branch|Tag ref The target gitlab project branch/tag.
+        # Param:  Hash config A hash containing addional package configuration.
+        # Returns: Composer::Package::Package
+        def load(project, ref, config)
+
+          config['version'] = parse_version(ref)
+
+          # export distribution package
+          # this works only on public repositories since composer can't handle gitlab oAuth yet.
+          if project.public?
+            config['dist'] = {
+              'url'  => [project.web_url, 'repository', 'archive.zip?ref=' + ref.name].join('/'),
+              'type' => 'zip',
+              'reference' => ref.target
+            }
+          end
+
+          # export source package
+          config['source'] = {
+            'url' => project.url_to_repo,
```

Comment:
Afaik it's meant to go this way, so the repos still require access and no-one can access the code.